Assets/Editor/GridMapCreator.cs

```diff
@@ -0,0 +1,379 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+public class GridMapCreator : EditorWindow
+{
+    #region Private Variables
+
+    private StyleManager styleManager;
+
+    private List<List<Node>> nodes;
+    private List<List<PartsScript>> parts;
+
+    private GUIStyle currentStyle;
+    private GUIStyle empty;
+    private Vector2 nodePos;
+    private Vector2 offset;
+    private Vector2 drag;
+    private Rect MenuBar;
+    private Rect MenuBar2;
+    private GameObject mapGO;
+
+    private bool isErasing;
+    private int buttonWidth = 80;
+    private int mapWidth = 16;
+    private int mapHeight = 9;
+    private int boundsX = 315;
+    private int boundsY = 175;
+
+    #endregion
+
+    #region SetUp
+
+    /// <summary>
+    /// Setting up access to the tool in GUI Window
+    /// </summary>
+    [MenuItem("Window/Level Editor Tool")]
+    private static void OpenWindow()
+    {
+        GridMapCreator window = GetWindow<GridMapCreator>();
+        window.titleContent = new GUIContent("Level Editor Tool");        
+    }
+
+    private void OnEnable()
+    {
+        SetupStyles();
+        SetUpNodesAndParts();
+        SetUpMap();
+    }
+
+    /// <summary>
+    /// Sets up the map game object that will hold all child objects once instantiated. 
+    /// </summary>
+    private void SetUpMap()
+    {
+        try
+        {
+            mapGO = GameObject.FindGameObjectWithTag("Map");
+            RestoreTheMap(mapGO);
+        }
+        catch (Exception e)
+        {
+            throw e;
+        }
+        if (mapGO == null)
+        {
+            mapGO = new GameObject("Map");
+            mapGO.tag = "Map";
+        }
+    }
+
+    /// <summary>
+    /// This function is used to show the instantiated game objects on the map, and corresponds with the map on GUI
+    /// </summary>
+    /// <param name="mapGO">The game object of the map, the parent that holds the child objects once instantiated</param>
+    private void RestoreTheMap(GameObject mapGO)
+    {
+        if (mapGO.transform.childCount > 0)
+        {
+            for (int i = 0; i < mapGO.transform.childCount; i++)
+            {
+                //Gets the row index of the child object @ index
```

Comment:
I'm a biger fan of statement //comment formatitng than 
//comment
statement
as it is more readable to me.